shared/native/notification-listeners.shared.js

```diff
@@ -4,64 +4,74 @@ import * as ConfigGen from '../actions/config-gen'
 let lastLoggedInNotifyUsername = null
 
 // We get a counter for badge state, if we get one that's less than what we've seen we toss it
 let lastBadgeStateVersion = -1
 
-export default function(dispatch: Dispatch, getState: () => Object, notify: any): IncomingCallMapType {
-  const throttledDispatch = throttle(action => dispatch(action), 1000, {
-    leading: false,
-    trailing: true,
-  })
-  return {
-    'keybase.1.NotifyBadges.badgeState': ({badgeState}) => {
-      if (badgeState.inboxVers < lastBadgeStateVersion) {
-        logger.info(
-          `Ignoring older badgeState, got ${badgeState.inboxVers} but have seen ${lastBadgeStateVersion}`
-        )
-        return
-      }
+const throttledDispatch = throttle((dispatch, action) => dispatch(action), 1000, {
+  leading: false,
+  trailing: true,
+})
 
-      lastBadgeStateVersion = badgeState.inboxVers
+export const sharedBadgeStateActions = (badgeState: Object, dispatch: Function): Array<any> => {
+  const actions = []
 
-      const conversations = badgeState.conversations
-      const totalChats = (conversations || []).reduce((total, c) => total + c.unreadMessages, 0)
-      const action = NotificationsGen.createReceivedBadgeState({badgeState})
-      if (totalChats > 0) {
-        // Defer this slightly so we don't get flashing if we're quickly receiving and reading
-        throttledDispatch(action)
-      } else {
-        // If clearing go immediately
-        throttledDispatch.cancel()
-        dispatch(action)
-      }
-    },
-    'keybase.1.NotifySession.loggedIn': ({username}, response) => {
-      lastBadgeStateVersion = -1
-      if (lastLoggedInNotifyUsername !== username) {
-        lastLoggedInNotifyUsername = username
-      }
+  if (badgeState.inboxVers < lastBadgeStateVersion) {
+    logger.info(
+      `Ignoring older badgeState, got ${badgeState.inboxVers} but have seen ${lastBadgeStateVersion}`
+    )
+    return actions
+  }
 
-      dispatch(ConfigGen.createBootstrap({}))
-      response.result()
-    },
-    'keybase.1.NotifySession.loggedOut': () => {
-      lastBadgeStateVersion = -1
-      lastLoggedInNotifyUsername = null
+  lastBadgeStateVersion = badgeState.inboxVers
 
-      // Do we actually think we're logged in?
-      if (getState().config.loggedIn) {
-        dispatch(LoginGen.createLogoutDone())
-      }
-    },
-    'keybase.1.NotifyTracking.trackingChanged': ({username, isTracking}) => {
-      dispatch(ConfigGen.createUpdateFollowing({username, isTracking}))
-    },
+  const conversations = badgeState.conversations
+  const totalChats = (conversations || []).reduce((total, c) => total + c.unreadMessages, 0)
+  const action = NotificationsGen.createReceivedBadgeState({badgeState})
+  if (totalChats > 0) {
+    // Defer this slightly so we don't get flashing if we're quickly receiving and reading
+    throttledDispatch(dispatch, action)
+  } else {
+    // If clearing go immediately
+    throttledDispatch.cancel()
+    actions.push(action)
```

Comment:
since this thing is actually getting dispatch i think it should just dispatch itself. We're doing this short term since this has this throttled dispatch, so to mix it dispatching and then also returning actions to dispatch is kinda confusing. Also we call this function and then don't actually dispatch these actions so they actually drop on the floor